components/skeleton/test/skeleton-group.vdiff.js

```diff
@@ -0,0 +1,237 @@
+import '../demo/skeleton-test-container.js';
+import '../demo/skeleton-test-heading.js';
+import { expect, fixture, html, waitUntil } from '@brightspace-ui/testing';
+
+const oneSkeletonFixture = html`
+	<d2l-skeleton-group-test-wrapper style="width: 330px">
+		<d2l-test-skeleton-container skeleton></d2l-test-skeleton-container>
+		<d2l-test-skeleton-container></d2l-test-skeleton-container>
+		<d2l-test-skeleton-container></d2l-test-skeleton-container>
+	</d2l-skeleton-group-test-wrapper>
+`;
+
+const noSkeletonFixture = html`
+	<d2l-skeleton-group-test-wrapper style="width: 330px">
+		<d2l-test-skeleton-container></d2l-test-skeleton-container>
+		<d2l-test-skeleton-container></d2l-test-skeleton-container>
+		<d2l-test-skeleton-container></d2l-test-skeleton-container>
+	</d2l-skeleton-group-test-wrapper>
+`;
+
+const nestedGroupsRemoveAllFixture = html`
+	<div style="width: 330px">
+		<d2l-skeleton-group-test-wrapper id="to-remove" skeleton>
+			<d2l-skeleton-group-test-wrapper id="to-remove" skeleton>
+				<d2l-test-skeleton-heading level="1">Heading 1</d2l-test-skeleton-heading>
+				<d2l-test-skeleton-box id="to-remove" skeleton></d2l-test-skeleton-box>
+				<d2l-test-skeleton-container></d2l-test-skeleton-container>
+			</d2l-skeleton-group-test-wrapper>
+			<d2l-skeleton-group-test-wrapper>
+				<d2l-test-skeleton-box></d2l-test-skeleton-box>
+			</d2l-skeleton-group-test-wrapper>
+		</d2l-skeleton-group-test-wrapper>
+	</div>
+`;
+
+describe('d2l-skeleton-group', () => {
+
+	it('all-skeleton', async() => {
+		const elem = await fixture(html`
+			<d2l-skeleton-group-test-wrapper style="width: 330px">
+				<d2l-test-skeleton-container skeleton></d2l-test-skeleton-container>
+				<d2l-test-skeleton-container skeleton></d2l-test-skeleton-container>
+				<d2l-test-skeleton-container skeleton></d2l-test-skeleton-container>
+			</d2l-skeleton-group-test-wrapper>
+		`);
+		await expect(elem).to.be.golden();
+	});
+
+	it('one-skeleton', async() => {
+		const elem = await fixture(oneSkeletonFixture);
+		await expect(elem).to.be.golden();
+	});
+
+	it('no-skeleton', async() => {
+		const elem = await fixture(noSkeletonFixture);
+		await expect(elem).to.be.golden();
+	});
+
+	it('make-one-skeleton', async() => {
+		const elem = await fixture(noSkeletonFixture);
+		const container = elem.querySelector('d2l-test-skeleton-container');
+		container.skeleton = true;
+		await container.updateComplete;
+		await expect(elem).to.be.golden();
+	});
+
+	it('make-one-not-skeleton', async() => {
+		const elem = await fixture(oneSkeletonFixture);
+		const container = elem.querySelector('d2l-test-skeleton-container');
+		container.skeleton = false;
+		await container.updateComplete;
+		await expect(elem).to.be.golden();
+	});
+
+	it('add-element', async() => {
+		const elem = await fixture(noSkeletonFixture);
+		elem.insertAdjacentHTML('beforeend', '<d2l-test-skeleton-container skeleton></d2l-test-skeleton-container>');
+		await waitUntil(() => elem.skeleton);
```

Comment:
`add-element` and `remove-element` flaked a bit I think because the communication of the new state isn't immediate. `waitUntil` should be pretty quick here.